lib/lightgbm/booster.rb

```diff
@@ -141,8 +141,21 @@ def num_trees
       out.read_int
     end
 
-    # TODO support different prediction types
-    def predict(input, start_iteration: nil, num_iteration: nil, **params)
+
+    # Make prediction for a new dataset.
+    # C-API: LGBM_BoosterPredictForMat.
+    #
+    # @param input [Array, Array<Array,Hash>, Hash{String => Numeric, String}, Daru::DataFrame, Rover::DataFrame] Input data
+    # @param start_iteration [Integer] Start index of the iteration to predict
+    # @param num_iteration [Integer] Number of iteration for prediction, <= 0 means no limit
+    # @param predict_type [Integer] What should be predicted
+    #     - C_API_PREDICT_NORMAL: normal prediction, with transform (if needed);
+    #     - C_API_PREDICT_RAW_SCORE: raw score;
+    #     - C_API_PREDICT_LEAF_INDEX: leaf index;
+    #     - C_API_PREDICT_CONTRIB: feature contributions (SHAP values)
```

Comment:
Using a single argument instead of 3 boolean args, like Python API does.

https://lightgbm.readthedocs.io/en/stable/pythonapi/lightgbm.Booster.html#lightgbm.Booster.predict

https://github.com/microsoft/LightGBM/blob/e057ae08e6bf6c6c84f276a127423fb145ca5fdb/python-package/lightgbm/basic.py#L1079-L1081

Let me know which you prefer. I don't really see the point of 3 booleans, since they are exclusive...